cocos/renderer/gfx-base/GFXDescriptorSet.cpp

```diff
@@ -40,6 +40,8 @@ DescriptorSet::~DescriptorSet() {
 }
 
 void DescriptorSet::initialize(const DescriptorSetInfo& info) {
+    CCASSERT(info.layout, "Invalid set layout");
```

Comment:
Why not add it in validate layer?

Reply:
because the base class needs the `info.layout` to be non-null right away

Reply:
Ok, got it.